waitress/server.py

```diff
@@ -77,6 +80,9 @@ def __init__(self,
         self.effective_host, self.effective_port = self.getsockname()
         self.server_name = self.get_server_name(self.adj.host)
         self.active_channels = {}
+        self.shutdown_gracefully = False
```

Comment:
make this __shutdown_gracefully, so users don't have direct access to this mechanism, but then add a shutdown() method that sets it, so users can install their own signal handling if needed.